server/src/test/java/org/elasticsearch/node/NodeTests.java

```diff
@@ -504,6 +605,26 @@ public void testAdditionalSettingsClash() {
         assertTrue(msg, msg.contains("plugin [" + AdditionalSettingsPlugin2.class.getName()));
     }
 
+    public void testPluginComponentInterfaceBinding() throws IOException, NodeValidationException {
+        List<Class<? extends Plugin>> plugins = basePlugins();
+        plugins.add(MockPluginWithAltImpl.class);
+        try (Node node = new MockNode(baseSettings().build(), plugins)) {
+            MockPluginWithAltImpl.MyInterface myInterface = node.injector().getInstance(MockPluginWithAltImpl.MyInterface.class);
+            MockPluginWithAltImpl plugin = node.getPluginsService().filterPlugins(MockPluginWithAltImpl.class).get(0);
+            if (plugin.getRandomBool()) {
+                assertThat(myInterface, instanceOf(MockPluginWithAltImpl.Foo.class));
+                assertTrue(myInterface.get().equalsIgnoreCase("foo"));
```

Comment:
I think assertThat should be used for all of these asserts, not just instanceOf.

Reply:
done (but left the actual boolean assertions to `assertTrue`) [`1b3560b` (#94347)](https://github.com/elastic/elasticsearch/pull/94347/commits/1b3560b8e0f0f741d7c799c5840757baa2c2e6ea)